third_party/terraform/data_sources/data_source_secretmanager_secret_version.go.erb

```diff
@@ -0,0 +1,122 @@
+<% autogen_exception -%>
+package google
+<% unless version == "ga" -%>
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceSecretmanagerSecretVersion() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceSecretmanagerSecretVersionRead,
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"secret": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: compareSelfLinkOrResourceName,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"create_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"destroy_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"secret_data": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func dataSourceSecretmanagerSecretVersionRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	fv, err := parseProjectFieldValue("secrets", d.Get("secret").(string), "project", d, config, false)
+	if err != nil {
+		return err
+	}
+	project := fv.Project
+	d.Set("secret", fv.Name)
+
+	var url string
+	versionNum := d.Get("version")
+
+	if versionNum != "" {
+		url, err = replaceVars(d, config, "{{SecretmanagerBasePath}}projects/{{project}}/secrets/{{secret}}/versions/{{version}}")
+		if err != nil {
+			return err
+		}
+	} else {
+		url, err = replaceVars(d, config, "{{SecretmanagerBasePath}}projects/{{project}}/secrets/{{secret}}/versions/latest")
+		if err != nil {
+			return err
+		}
+	}
+
+	var version map[string]interface{}
+	version, err = sendRequest(config, "GET", project, url, nil)
+	if err != nil {
+		return fmt.Errorf("Error retrieving available secret manager secret versions: %s", err.Error())
+	}
+
+	secretVersionRegex := regexp.MustCompile("projects/(.+)/secrets/(.+)/versions/(.+)$")
+
+	parts := secretVersionRegex.FindStringSubmatch(version["name"].(string))
+	// should return [full string, project number, secret name, version number]
+	if len(parts) != 4 {
```

Comment:
There is no way this should fail, right?

`version["name"]` is coming back from the API, not user input. If it isn't in the correct format then something has gone horribly wrong and the user cannot fix it

Reply:
I believe so.  Does this mean we don't want the error here?  I just had it in as a just-in-case since we access `parts[3]` later.

Reply:
It's probably fine. It would be confusing for a user to see this error and think that they need to fix it, and a panic might be more obvious that the error is in the provider (or API), but I don't have strong opinions and checking before array access seems like a good idea